STATE_TRANSITIONS.MD

```diff
@@ -0,0 +1,224 @@
+# Run State Transitions
+Run states change based on the success or failure of a job run. The datasets consumed and/or produced
+by a job run are immutable and do not change based on the success or failure of the run.
+
+# Dataset State Transitions
+Dataset states change based on the success or failure of job runs that consume or generate those 
+datasets. Typically, Runs generate dataset versions as outputs, but sometimes a job run may result 
+in an input dataset version being created if that dataset did not previously exist in the database.
+Once a dataset exists, a new version will only be created if the dataset is the output of a job run. 
```

Comment:
The cases bellow expose all the possible paths taken in an algorithm.
Could you add a summary at the top with the pseudo code? That will make it easier to follow each of the individual paths bellow.
I would expect a few nested if-else that describe this logic.